src/pluginplay/printing/mermaid.cpp

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2024 NWChemEx-Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "module/module_class.hpp"
+#include "module_manager/module_manager_class.hpp"
+#include <pluginplay/pluginplay.hpp>
+
+void print_submods(const std::string module,
+                   const pluginplay::ModuleManager& mm, int level = 0) {
+    auto mm_module = mm.at(module);
```

Comment:
```suggestion
    const auto& mm_module = mm.at(module);
```
TL;DR you were copying the module, this will avoid the copy. Best to get in the habit of avoiding copies.

Full answer, C++ is fun 😝 and mastering it requires memorizing a gazillion seamingly orthogonal rules. One of those rules is that `auto` follows template type parameter deduction rules, which I'm assuming you haven't learned (completely understandable at this point in your C++ journey). So what you need to remember when using `auto` is that 90% of the time `auto` will deduce a value type and copy the object, meaning you usually want `auto&` or `const auto&` (unless you really want the copy). What makes this fun is this happens even though `ModuleManager::at` returns `Module&`. This is even more fun because if we had written `ModuleManager::at` incorrectly so that it returns `Module`, then if you blindly apply the `auto&`/`const auto&` rule you'd take a reference to a temporary and likely got a segfault. That all said, `auto` is great. You should be using it, but it's not a true black box. You still have to understand the API of what you're interacting with.